src/RadioButton.tsx

```diff
@@ -0,0 +1,76 @@
+import styled from 'styled-components'
+import React, {InputHTMLAttributes, ReactElement} from 'react'
+import sx, {SxProp} from './sx'
+
+export type RadioButtonProps = {
+  /**
+   * A unique value that is never shown to the user.
+   * Used during form submission and to identify which radio button in a group is selected
+   */
+  value: string
+  /**
+   * Name attribute of the input element. Required for grouping radio inputs
+   */
+  name: string
+  /**
+   * Apply inactive visual appearance to the radio button
+   */
+  disabled?: boolean
+  /**
+   * Indicates whether the radio button is selected
+   */
+  checked?: boolean
+  /**
+   * Forward a ref to the underlying input element
+   */
+  ref?: React.RefObject<HTMLInputElement>
+  /**
+   * Indicates whether the radio button must be checked before the form can be submitted
+   */
+  required?: boolean
+  /**
+   * Indicates whether the radio button validation state is non-standard
+   */
+  validationStatus?: 'error' | 'success' // TODO: hoist to Validation typings
+} & InputHTMLAttributes<HTMLInputElement> &
+  SxProp
+
+const StyledRadioButton = styled.input`
+  cursor: pointer;
+
+  ${props => props.disabled && `cursor: not-allowed;`}
+
+  ${sx}
+`
+
+/**
+ * An accessible, native radio component for selecting one option from a list.
+ */
+const RadioButton = React.forwardRef<HTMLInputElement, RadioButtonProps>(
+  (
+    {checked, disabled, sx: sxProp, required, validationStatus, value, name, ...rest}: RadioButtonProps,
+    ref
+  ): ReactElement => {
+    return (
+      <StyledRadioButton
+        type="radio"
+        value={value}
+        name={name}
+        ref={ref}
+        disabled={disabled}
+        aria-disabled={disabled ? 'true' : 'false'}
```

Comment:
Wondering if we need to set the aria- properties at all? Taking care of the main html properties should be enough? Keen to learn more.

Reply:
It's a good question. They are primarily here to provide additional context for screen readers.. but also:

- Some screen readers supposedly have [better support ](https://www.levelaccess.com/how-not-to-misuse-aria-states-properties-and-roles/)for aria attributes vs inferring values from semantic tags. E.g. `aria-required`
- Some aria roles aren't supported natively. E.g. `aria-invalid`. So adding those should hopefully provide screen readers with additional context on UI state 
- Some aria tags have better support for non-standard values.  E.g.  `aria-checked`, which supports `mixed` to represent a js-only indeterminate modes.
- Explicitly setting a string boolean value felt like a stricter implementation of html, which should provide better utility while writing css or tests, rather than checking for whether an aria attribute is set or not by the js framework
- There is test coverage added against aria roles to guard the component API against future breaking changes /  refactoring, in case we were ever to move away from native checkboxes and radios in favour of custom ones.

Admittedly, there are a lot of assumptions here though. For some of these aria attributes, it may be better to remove them. @alliethu - do you have any opinions on this too?

Reply:
@rezrah As long as you're using **native** HTML radio buttons, you should not need to include these additional ARIA roles. The ARIA roles are made to be used with custom radio buttons to mimic the native version.